mixes/getters.js

```diff
@@ -12,14 +14,20 @@ const getMixesError = (state) => state.mixes.error
 
 const getMixes = Getter(
   getMixesRecords,
+  getMetas,
   getChannels,
   getClips,
-  (mixes, channels, clips) => {
-    return mapValues(mixes, (mix, mixId) => {
-      return (mix.channelId && channels[mix.channelId])
-        ? nestMix({ ...mix, channels, clips })
-        : mix
-    })
+  (mixes, metas, channels, clips) => {
+    return omitBy(mapValues(mixes, (flatMix, mixId) => {
+      if (channels[flatMix.channelId]) {
+        const nestedMix = nestMix({ ...flatMix, channels, clips })
+        return {
+          ...nestedMix,
+          meta: metas[nestedMix.id],
+          primaryTracks: getPrimaryTracks(nestedMix, metas)
+        }
+      }
+    }), isNil)
```

Comment:
couple of changes here... essentially, we only ever want this to be a list of nested mixes. so we filter out the flat mixes which do not have enough data to be nested. again with loop effect race conditions and defensiveness, there is a case where a mix is loaded but its channels havent been set yet